tools/smithy-rs-sync/src/main.rs

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use anyhow::{anyhow, bail, Context, Result};
+use git2::{Commit, IndexAddOption, ObjectType, Oid, Repository, ResetType, Signature};
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::Instant;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "smithy-rs-sync")]
+/// A CLI tool to replay commits from smithy-rs, generate code, and commit that code to aws-rust-sdk
```

Comment:
Nit: most comments / doc are missing the carriage return.

Reply:
Do you mean that I should have split this comment into 2 lines?

Reply:
Nope, they miss the "." at the end of the line. Again, nitpicking here, not a blocker.

Reply:
I have since added these